### ISSUE TITLE
Maya Tool - Quickly Bake Attributes Tool

### ISSUE THREAD

**Issue by david-cattermole** [david-cattermole/mayaMatchMoveSolver#184]
### Feature

Users want to quickly bake the attributes of an object.

We want to automatically disable the viewport, and switch to DG Evaluation to speed up baking (using mmSolver API).

We have the choice of baking the selected attributes in the channel box, or all attributes on the selected objects.

This should also support custom frame ranges, allowing the "inner" frame range, the "outer" frame range or a custom user-entered frame range.

We also need to include a Smart Bake checkbox - to enable/disable using smart (sparse keyframes) baking.

This tool will need an Option Box window, and should have a "Reset" and "Bake" button. The Reset button will reset the attribute. The options are saved inside the Maya scene file.

Also we can display the time taken to bake at the end of the baking process:
```python
import time
s = time.time()
time.sleep(1)
e = time.time()
LOG.warn('Time elapsed: %r secs', e - s)
```

### Usage Steps

1. Select object(s)
2. Select the attribute(s) in the channel box.
3. Run tool
    - The viewport will be disabled, and re-enabled after baking is complete.
    - The DG mode will be forced to DG Evaluation, then re-enabled after baking is complete
    - The attributes will be keyframed over time - if smart bake is enabled then the keyframes are "sparsely baked"
    - Any rotation keyframes will be "euler filtered".
    - At the end of the bake, the time is measured and displayed to the user (with `LOG.warn()`).

### Software Versions

- mmSolver version: Latest development version
- Maya version: All supported
- Operating System (OS): All supported

**Comment by bpatchasaheb** [comment]
I would like to develop this tool.

**Comment by david-cattermole** [comment]
Ok, thanks for helping out!

@ktonegawa I just wanted to make sure you're not doing anything on this - I don't think you are?

**Comment by ktonegawa** [comment]
Hii @david-cattermole, no I have not done anything for this. Looking forward to seeing what @bpatchasaheb comes up with.

**Comment by david-cattermole** [comment]
This tool has just been merged in #192.

The last steps will be to write documentation.

**Comment by david-cattermole** [comment]
I have added some documentation:
https://github.com/david-cattermole/mayaMatchMoveSolver/blob/develop_v0.3.x/docs/source/tools_attributetools.rst#bake-attributes

This is now ready for release and will be included in v0.3.13.
